rpcserver.go

```diff
@@ -1251,3 +1298,62 @@ func marshallSendAssetEvent(
 		return nil, fmt.Errorf("unknown event type: %T", eventInterface)
 	}
 }
+
+// unmarshalScriptKey parses the RPC script key into the native counterpart.
+func unmarshalScriptKey(rpcKey *tarorpc.ScriptKey) (asset.ScriptKey, error) {
+	var (
+		scriptKey asset.ScriptKey
+		err       error
+	)
+	scriptKey.PubKey, err = schnorr.ParsePubKey(rpcKey.PubKey)
+	if err != nil {
+		return scriptKey, err
+	}
+
+	// The key descriptor is optional for script keys that are completely
+	// independent of the backing wallet.
+	if rpcKey.KeyDesc != nil {
+		keyDesc, err := unmarshalKeyDescriptor(rpcKey.KeyDesc)
+		if err != nil {
+			return scriptKey, err
```

Comment:
nit: `return nil, err` (along with function signature changes, etc.) would be clearer here, IMO.